hack/test-go.sh

```diff
@@ -101,6 +104,11 @@ for arg; do
     testcases+=("${arg}")
   fi
 done
+# If either no test cases were provided, or if the first test case is an empty
+# string (needed to specify Go arguments?), run all tests.
```

Comment:
a few places were using "" as the first argument to bypass getopt intercepting Go args.
I updated the callers to use -- instead and removed this special case.